lib/broccoli/ember-app.js

```diff
@@ -1658,20 +1711,22 @@ class EmberApp {
           if (!entry.transformation) {
             throw new Error(`while importing ${assetPath}: each entry in the \`using\` list must have a \`transformation\` name`);
           }
-          if (entry.transformation !== 'amd') {
-            throw new Error(`while importing ${assetPath}: unknown transformation \`${entry.transformation}\``);
-          }
-          if (!entry.as) {
-            throw new Error(`while importing ${assetPath}: amd transformation requires an \`as\` argument that specifies the desired module name`);
-          }
-          if (!this.amdModuleNames) {
-            this.amdModuleNames = {};
+
+          let transformName = entry.transformation;
+
+          if (this._customTransformOrder.indexOf(transformName) === -1) {
+            let availableTransformNames = this._customTransformOrder.join(',');
+            throw new Error(`while import ${assetPath}: found an unknown transformation name ${transformName}. Available transformNames are: ${availableTransformNames}`);
           }
-          // If the import is specified to be a different name we must break because of the broccoli rewrite behavior.
-          if (this.amdModuleNames[assetPath] && this.amdModuleNames[assetPath] !== entry.as) {
-            throw new Error(`Highlander error while importing ${assetPath}. You may not import an AMD transformed asset at different module names.`);
+
+
+          if (this._customTransforms[transformName].processOptions) {
```

Comment:
I'd prefer to just default `processOptions` to `(options) => options;`, which allows us to avoid the extra branching and simplifies this a tiny bit.